### PR TITLE
Add WebAssembly example snap

### PR DESCRIPTION
**Pull request by rekmarks**
Adds WebAssembly example snap which compiles AssemblyScript to WebAssembly and uses it in a snap. This is the first iteration, later on we can experiment with showing off bindings etc.

### CODE REVIEW

File: packages/snap-examples/examples/wasm/src/index.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    console.error('Failed to initialize WebAssembly module.', error);
+    throw error;
+  }
+};
+
+wallet.registerRpcMessageHandler(async (_originString, requestObject) => {
+  if (!wasm) {
+    await initializeWasm();
+  }
+
```
Comment (shanejonas) [inline packages/snap-examples/examples/wasm/src/index.js:60]:
```suggestion
  const { method, params } = requestObject;
  
  if (asm.instance.exports[method]) {
    return asm.instance.exports.[method](...params);
  }
```

Comment (shanejonas) [reply packages/snap-examples/examples/wasm/src/index.js:60]: didnt try this but would be awesome if it worked like this. not sure how the params would map to wasm types

Comment (shanejonas) [reply packages/snap-examples/examples/wasm/src/index.js:60]: looks like it should work:

image
> source: https://rob-blackbourn.github.io/blog/webassembly/wasm/array/arrays/javascript/c/2020/06/07/wasm-arrays.html

**Review by ritave — DISMISSED** [review]
If we have AssemblyScript source code, let's remove `program.wasm` and compile it during build time.

Additionally - I'm currently rewriting cli to use rollup instead of browserify. Rollup has a working wasm plugin, so you might want to wait for that to get in before merging this PR

File: packages/examples/examples/wasm/src/index.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    throw error;
+  }
+};
+
+wallet.registerRpcMessageHandler(async (_originString, requestObject) => {
+  if (!wasm) {
+    await initializeWasm();
+  }
+
+  switch (requestObject.method) {
```
Comment (shanejonas) [inline packages/examples/examples/wasm/src/index.js:25]: we should just pass the method/params through to be called on wasm if it exists:

```
  if (wasm.instance.exports[requestObject.method]) {
    return wasm.instance.exports[requestObject.method](...requestObject.params);
  }
```

Comment (FrederikBolding) [reply packages/examples/examples/wasm/src/index.js:25]: Sure 👍

Comment (FrederikBolding) [reply packages/examples/examples/wasm/src/index.js]: Done!

**Comment by rekmarks** [comment]
@ritave your request to compile to `.wasm` from program source code was added, and we'll update this example to use Rollup once we've landed that.

**Comment by rekmarks** [comment]
I approve of this PR but I authored it so someone else has to approve, cc: @FrederikBolding

**Review by shanejonas — APPROVED** [review]

**Review by hmalik88 — APPROVED** [review]

**Pull request merged**